staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -633,6 +636,21 @@ func (a *Authenticator) AuthenticateToken(ctx context.Context, token string) (*a
 		}
 	}
 
+	// Ensure not-before (nbf) claim, if present, is indeed in the past.
+	if c.hasClaim("nbf") {
+		var nbfValue int64
+		if err := c.unmarshalClaim("nbf", &nbfValue); err != nil {
+			return nil, false, fmt.Errorf("oidc: parse claim nbf: %v", err)
+		}
+
+		nbfTime := time.Unix(nbfValue, 0)
+		now := a.now()
```

Comment:
now() can be nil here. This needs to default to time.Now above.

Reply:
Ah, alright - golang new to me so still trying to figure out all the conditions for when nil checks are needed. I thought the check at line 308 would be enough, but coming to think of it of course that only covers the case where the authenticator is constructed by the `newAuthenticator` function. Any good tools for finding possible nil-problems like there is for Java? I'll fix.

Reply:
I don't know of a common one. Generally, you ensure this during initialization:

```
now := config.now
if now == nil {
    now = time.Now
}
a := authenticator{
    now: now,
    // ...
}
```

Reply:
Or  `a := authenticator{now: time.Now}` and set over it in tests.